drivers/rtc/rtc_ll_stm32.c

```diff
@@ -208,10 +188,22 @@ static int rtc_stm32_set_time(const struct device *dev, const struct rtc_time *t
 	}
 
 	LOG_INF("Setting clock");
+
+#if defined(PWR_CR_DBP) || defined(PWR_CR1_DBP) || defined(PWR_DBPCR_DBP) || defined(PWR_DBPR_DBP)
+	LL_PWR_EnableBkUpAccess();
+#endif /* PWR_CR_DBP || PWR_CR1_DBP || PWR_DBPR_DBP */
+
 	LL_RTC_DisableWriteProtection(RTC);
 
-	err = rtc_stm32_enter_initialization_mode(true);
+	ErrorStatus status = LL_RTC_EnterInitMode(RTC);
+
+	if (status != SUCCESS) {
+		err = -EIO;
+	}
 	if (err) {
```

Comment:
It may be better to merge the two "if" ?

Reply:
I agree, updated.